system-addon/lib/DownloadsManager.jsm

```diff
@@ -98,17 +96,31 @@ this.DownloadsManager = class DownloadsManager {
     }
   }
 
-  get downloads() {
+  async getDownloads({threshold = 0, numItems = this._viewableDownloadItems.size, onlyExists = false}) {
     let results = [];
-    for (const url of this._viewableDownloadItems.keys()) {
-      const elem = this._viewableDownloadItems.get(url);
-      // We only want downloads that are less than 36 hours old
-      if (elem && (elem.downloadedSince < RECENT_DOWNLOAD_THRESHOLD)) {
-        const formattedDownloadForHighlights = this.formatDownload(elem);
-        results.push(formattedDownloadForHighlights);
+    for (const elem of this._viewableDownloadItems.values()) {
+      // Only get downloads within the time threshold specified
+      if (elem && (elem.downloadedSince < threshold)) {
```

Comment:
Also just documenting from irc. `elem` shouldn't ever be falsey now. And this comparison expands to `Date.now() - endTime < threshold` and gets computed for every download. We can shuffle things around to be `Date.now() - threshold < endTime` which makes the math loop invariant, so compute a `dateThreshold = Date.now() - threshold` once before the loop and check `endTime > dateThreshold`